src/systemd/sd-bus-vtable.h

```diff
@@ -61,13 +61,16 @@ struct sd_bus_vtable {
                 struct {
                         const char *member;
                         const char *signature;
+                        const char *in_names;
                         const char *result;
+                        const char *out_names;
```

Comment:
these fields must appear at the end of the structure. That's because these structs show up as part of the sd-bus public ABI, and we'd break ABI if we'd shift existing fields around. The start.element_size field (which is defined in the first vtable item) will let us know if the extra field in the struct is defined or not.

Also, to keep the size of structure small, it might be a good idea, to add only one field here, and call it "parameter_names", and then simply say the the input names need to be listed first and the output names second. The macro putting these structs together otoh should take both lists separately, and just concatenate them here implicitly.

Reply:
ok, and what about also a macro
            #define SD_BUS_NAME(_name) #_name "\0"
just to help the developer make sure the \0 is always appended, and see visually more names:
    SD_BUS_NAME(firstname) SD_BUS_NAME(lastname) SD_BUS_NAME(city)
?

Reply:
hmm, but this should really be called SD_BUS_PARAMETER(x) or so, "name" is just too generic. Or SD_BUS_PARAM() if you must...